test/functional/lib/redirect-handler/blockOrObserve.test.ts

```diff
@@ -30,9 +30,41 @@ const dynamicRulesConditions = (regexFilter) => ({
   ]
 })
 
+const TEST_TAB_ID = 1234
+const LAST_GROUP_REGEX = '((?:[^\\.]|$).*)$'
+
+/**
+ * Ensures that the tab is redirected to the given url on the first request.
+ *
+ * @param url
+ */
+function ensureTabRedirected (url): void {
+  expect(browserMock.tabs.query.called).to.be.true
+  expect(browserMock.tabs.update.called).to.be.true
+  expect(browserMock.tabs.update.firstCall.args).to.deep.equal([TEST_TAB_ID, { url }])
```

Comment:
should we accept a `TEST_TAB_ID` argument? I imagine we will need to test multiple open tabs at some point?